### PR TITLE
Sockets Tatiana

### PR DESCRIPTION
**Pull request by tatsqui**
# Media Ranker
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
---      | ---
Describe a custom model method you wrote. | The spotlight method - it prefers the work with the most upvotes. 
Describe how you approached testing that model method. What edge cases did you come up with? | I did three different tests, one test based on the fixture data that selects the fixture work with the most votes (nominal case). The edge cases: In the case of a tie, it just chooses whichever work, and if there are no votes, it selects whatever vote (won't be nil). 
What are `session` and `flash`? What is the difference between them? | They are both hash like objects that we can use in different ways. Flash can be used to display a message to the client either on render or send a message that persists one cycle on redirect. Session persists until the user closes their browser, and retrieves the data it is assigned and stores that data in a cookie. 
What was one thing that you gained more clarity on through this assignment? | I gained more clarity on writing tests for Rails! Also, I gained more clarity on how this framework works, which I think is pretty cool and I enjoyed working on this project. 
What is the Heroku URL of your deployed application? | https://mediaranker-tq.herokuapp.com/
Do you have any recommendations on how we could improve this project for the next cohort? | A mid-point checkin on how the project is coming along (like we did with hotel) I think could be useful for folks who get off to slow starts(like me).

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Rails.application.routes.draw do
+  root to: "homepages#index"
+
+  resources :works do
+    resources :upvotes, only: [:create]
```
Comment (dHelmgren) [inline config/routes.rb:5]: nice work keeping this dry!

File: app/views/works/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,122 @@
+<div class="alert"> TODO: ADD LINKS TO UPVOTE AND ACCURATE VOTE COUNT </div>
```
Comment (dHelmgren) [inline app/views/works/index.html.erb:1]: Uh oh!

File: app/views/works/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+      <% end %>
+    </table>
+  <% else %>
+    No media available
+  <% end %>
+</section>
+
+<section class="media-table">
+  <h4> Albums </h4>
+  <% if !Work.list("album").empty? %>
```
Comment (dHelmgren) [inline app/views/works/index.html.erb:45]: You have this code repeated 3 times. Use a view partial or view helper to dry this up!

**Comment by dHelmgren** [comment]
# Media Ranker
## What We're Looking For

Feature | Feedback
---     | ---
**Core Requirements** |
Git hygiene | Good
Comprehension questions | yes
**General** |
Rails fundamentals (RESTful routing, use of named paths) |
Views are well-organized (DRY, use of semantic HTML, use of partials) |
Errors are reported to the user |
Business logic lives in the models |
Models are thoroughly tested, including relations, validations and any custom logic |
Controllers are thoroughly tested, including the login/logout process and multi-step workflows like voting for a work |
**Wave 1 - Media** |
Splash page shows the three media categories | yes
Basic CRUD operations on media are present and functional | Yes, but you've got some TODO's lying around
**Wave 2 - Users and Votes** |
Users can log in and log out | yes
The ID of the current user is stored in the session | yes
A user cannot vote for the same media more than once | yes
All media lists are ordered by vote count | yes
Splash page contains a media spotlight | yes
**Wave 3 - Users and Votes** |
Media pages contain lists of voting users | yes
Individual user pages and the user list are present | yes
**Optional - Styling** |
Bootstrap is used appropriately | yes
Look and feel is similar to the original | yes
**Overall** | Great job overall! Your implementation matches the demo site very closely, and I would say the learning goals for this assignment were largely met. There are a few places where things could be cleaned up, which I've tried to outline below, but in general I am quite happy with this submission. Make sure you double check your TODO tags!
